### PR TITLE
AWS service guide

### PR DESCRIPTION
**Pull request by soorena776**
Signed-off-by: soorena776 <javad@upbound.io>

### Checklist

I have:
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Ensured this PR contains a neat, self documenting set of commits.
- [ ] Updated any relevant [documentation], [examples], or [release notes].
- [ ] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

[documentation]: https://github.com/crossplaneio/crossplane/tree/master/docs
[examples]: https://github.com/crossplaneio/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplaneio/crossplane/tree/master/PendingReleaseNotes.md
[`clusterrole.yaml`]: https://github.com/crossplaneio/crossplane/blob/master/cluster/charts/crossplane/templates/clusterrole.yaml

### CODE REVIEW

File: docs/services/services-guide-aws.md [hunk 1]
```diff
@@ -0,0 +1,827 @@
...
+
+This guide assumes that you have setup and configured the AWS CLI.
+
+*Note: these session variables are used throughout this guide. You may use the
+values below or create your own.*
+
+
+### Set Up an EKS Cluster
+
+We will create an EKS cluster, follwoing the steps provided in [AWS documentation] [aws-create-eks].
```
Comment (negz) [inline docs/services/services-guide-aws.md]: The `[AWS documentation]` link is dangling

File: docs/services/services-guide-aws.md [hunk 2]
```diff
@@ -0,0 +1,827 @@
...
+# comma separated list of Subnet IDs
+SUBNET_IDS=$(aws cloudformation describe-stacks --output json --stack-name ${EKS_STACK_NAME} --region ${REGION} | jq -r '.Stacks[0].Outputs[]|select(.OutputKey=="SubnetIds").OutputValue')
+
+# the ID of security group that later will be used for the EKS cluster
+EKS_SECURITY_GROUP=$(aws cloudformation describe-stacks --output json --stack-name ${EKS_STACK_NAME} --region ${REGION} | jq -r '.Stacks[0].Outputs[]|select(.OutputKey=="SecurityGroups").OutputValue')
+```
+
+#### Create an IAM Role for the EKS cluster
+
+For EKS cluster to be able to access different resources, it needs to be given the required permissions through an **IAM Role**. In this section we create a role and assign the required policies. Later we will make EKS to assume this role.
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
For the EKS cluster to be able to access different resources, it needs to be given the required permissions through an **IAM Role**. In this section we create a role and assign the required policies. Later we will make EKS assume this role.
```

File: docs/services/services-guide-aws.md [hunk 3]
```diff
@@ -0,0 +1,827 @@
...
+
+The output should be the created role in JSON. Next, you'll attach the required policies to this role:
+
+```bash
+# attach the required policies
+aws iam attach-role-policy --role-name "${EKS_ROLE_NAME}" --policy-arn arn:aws:iam::aws:policy/AmazonEKSClusterPolicy > /dev/null
+aws iam attach-role-policy --role-name "${EKS_ROLE_NAME}" --policy-arn arn:aws:iam::aws:policy/AmazonEKSServicePolicy > /dev/null
+```
+
+Now lets retrieve the **arn** of this role and store it in a variable. We later assign this to the EKS.
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
Now lets retrieve the **ARN** of this role and store it in a variable. We later assign this to the EKS.
```

File: docs/services/services-guide-aws.md [hunk 4]
```diff
@@ -0,0 +1,827 @@
...
+
+aws eks create-cluster \
+  --region "${REGION}" \
+  --name "${CLUSTER_NAME}" \
+  --kubernetes-version "${K8S_VERSION}" \
+  --role-arn "${EKS_ROLE_ARN}" \
+  --resources-vpc-config subnetIds="${SUBNET_IDS}",securityGroupIds="${EKS_SECURITY_GROUP}"
+```
+
+At this point, an EKS cluster should be started provisioning, which could take take up to 15 minutes. You can check the status of the cluster by running:
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
At this point, an EKS cluster should have started provisioning, which could take take up to 15 minutes. You can check the status of the cluster by running:
```

File: docs/services/services-guide-aws.md [hunk 5]
```diff
@@ -0,0 +1,827 @@
...
+
+```bash
+aws eks describe-cluster --name "${CLUSTER_NAME}" --region "${REGION}" | jq -r .cluster.status
+```
+
+Once the provisioning is completed, the above command will return `ACTIVE`.
+
+#### Configuring `kubectl` to communicatie with the EKS
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
#### Configuring `kubectl` to communicate with the EKS cluster
```

File: docs/services/services-guide-aws.md [hunk 6]
```diff
@@ -0,0 +1,827 @@
...
+
+```bash
+aws eks describe-cluster --name "${CLUSTER_NAME}" --region "${REGION}" | jq -r .cluster.status
+```
+
+Once the provisioning is completed, the above command will return `ACTIVE`.
+
+#### Configuring `kubectl` to communicatie with the EKS 
+
+Once the cluster is created and is `ACTIVE`, we configure the `kubectl` to target this cluster:
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
Once the cluster is created and is `ACTIVE`, we configure `kubectl` to target it:
```

File: docs/services/services-guide-aws.md [hunk 7]
```diff
@@ -0,0 +1,827 @@
...
+
+Which should produce something like:
+
+>```bash
+>Kubernetes master is running at https://12E34567898A607F40B3C2FDDF42DC5.sk1.eu-west-1.eks.amazonaws.com
+>```
+
+#### Creating Worker Nodes for the EKS cluster
+
+The worker noes of an EKS cluster are not managed by the cluster itself. Instead, a set of EC2 instances, along with other resources and configurations are needed to be provisioned. In this section, we will create another Cloud Formation stack to setup a worker node configuration, which is described in [EKS official documentation][sample-workernodes-stack].
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
The worker nodes of an EKS cluster are not managed by the cluster itself. Instead, a set of EC2 instances, along with other resources and configurations need to be provisioned. In this section, we will create another Cloud Formation stack to setup a worker node configuration, which is described in [EKS official documentation][sample-workernodes-stack].
```

File: docs/services/services-guide-aws.md [hunk 8]
```diff
@@ -0,0 +1,827 @@
...
+
+The output will look like:
+
+>```bash
+>{
+>    "StackId": "arn:aws:cloudformation:eu-west-1:123456789012:stack/aws-serivce-example-stack 5730d720-d9d9-11e9-8662-029e8e947a9c"
+>}
+>```
+
+Similar to VPC stack, creating the workers stack continutes in the background and could take a few minutes to complete. You can check its status by running:
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
Similar to the VPC stack, creating the workers stack continues in the background and could take a few minutes to complete. You can check its status by running:
```

File: docs/services/services-guide-aws.md [hunk 9]
```diff
@@ -0,0 +1,827 @@
...
+> ip-192-168-104-194.eu-west-1.compute.internal   NotReady   <none>   8s    v1.14.6-eks-5047ed
+>```
+
+After a few minutes, the `NotReady` status above should become `Ready`.
+
+Congratulations! You have successfully setup and configured your EKS cluster!
+
+### Set Up RDS Configurations
+
+In AWS an RDS database instance, will be provisioned to satisfy WordPress application's `MySQLInstanceClass` claim. In order to make an RDS instance accessible by the EKS cluster:
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
In AWS an RDS database instance will be provisioned to satisfy WordPress application's `MySQLInstanceClass` claim. In order to make an RDS instance accessible by the EKS cluster:
```

File: docs/services/services-guide-aws.md [hunk 10]
```diff
@@ -0,0 +1,827 @@
...
+
+1. A security group should be created and assigned to the RDS, so certain traffic from the EKS is allowed
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
1. A security group should be created and assigned to the RDS, so certain traffic from the EKS cluster is allowed
```

File: docs/services/services-guide-aws.md [hunk 11]
```diff
@@ -0,0 +1,827 @@
...
+  --vpc-id="${VPC_ID}" \
+  --region="${REGION}" \
+  --group-name="${RDS_SG_NAME}" \
+  --description="open mysql access for crossplane-example cluster" 
+
+# retrieve the ID for this security group
+RDS_SECURITY_GROUP_ID=$(aws ec2 describe-security-groups --filter Name=group-name,Values="${RDS_SG_NAME}" --region="${REGION}" --output=text --query="SecurityGroups[0].GroupId")
+```
+
+After creating the security group, we add a rule to allow traffic at `MySQL` port
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
After creating the security group, we add a rule to allow traffic on `MySQL`'s port
```

File: docs/services/services-guide-aws.md [hunk 12]
```diff
@@ -0,0 +1,827 @@
...
+
+aws rds create-db-subnet-group \
+  --region="${REGION}" \
+  --db-subnet-group-name="${DB_SUBNET_GROUP_NAME}" \
+  --db-subnet-group-description="crossplane-example db subnet group" \
+  --subnet-ids $SUBNETS_LIST > /dev/null
+
+```
+
+These resources later will be used to create cloud-specitic MySQL resources.
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
These resources will later be used to create cloud-specific MySQL resources.
```

File: docs/services/services-guide-aws.md [hunk 13]
```diff
@@ -0,0 +1,827 @@
...
+
+### Set Up Crossplane
+
+Using the newly provisioned cluster:
+
+1. Install Crossplane from alpha channel. (See the [Crossplane Installation
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
1. Install Crossplane from the alpha channel. (See the [Crossplane Installation
```

File: docs/services/services-guide-aws.md [hunk 14]
```diff
@@ -0,0 +1,827 @@
...
+metadata:
+  name: aws-infra-dev
+EOF
+
+kubectl apply -f aws-infra-dev-namespace.yam
+```
+
+* You should see the following output:
+
+> namespace/aws-infra-dev.yaml created
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
> namespace/aws-infra-dev created
```

(I think?)

File: docs/services/services-guide-aws.md [hunk 15]
```diff
@@ -0,0 +1,827 @@
...
+
+#### AWS Provider
+
+It is essential to make sure that the AWS user credentials is configured
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
It is essential to make sure that the AWS user credentials are configured
```

File: docs/services/services-guide-aws.md [hunk 16]
```diff
@@ -0,0 +1,827 @@
...
+in Crossplane as a provider. Please follow the steps [provider guide][aws-provider-guide]
+for more information.
+
+#### Cloud-Specific Resource Classes
+
+Cloud-specific resource classes are used to define a reusable configuration for
+a specific managed service. Wordpress requires a MySQL database, which can be
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
a specific managed resource. Wordpress requires a MySQL database, which can be
```

I think we more commonly say managed resource (meaning the actual `RDSInstance`), but managed service makes sense here too if you mean for our RDS support in general.

File: docs/services/services-guide-aws.md [hunk 17]
```diff
@@ -0,0 +1,827 @@
...
+* Created a `RDSInstanceClass` in the ` with configuration using RDS configuration earlier
+* Created a `MySQLInstanceClass` that specified the `RDSInstanceClass` as
+  `mysql-standard` in the `app-project1-dev` namespace
+* Created a `MySQLInstance` claim in the `app-project1-dev1` namespace that
+  referenced `mysql-standard`
+* Created a `Deployment` and `Service` to run Wordpress on our EKS Cluster and
+  assign an external IP address to it
+
+If you would like to try out a similar workflow using a different cloud
+provider, take a look at the other [services guides][serives]. If you would like
```
Comment (negz) [inline docs/services/services-guide-aws.md]:
```suggestion
provider, take a look at the other [services guides][services]. If you would like
```

File: docs/services/services-guide-aws.md [hunk 18]
```diff
@@ -0,0 +1,827 @@
+# Deploying Wordpress in Amazon Web Services (AWS)
```
Comment (displague) [inline docs/services/services-guide-aws.md]: This is missing front-matter to include it in the ToC:

```yaml
---
title: Using AWS Services
toc: true
weight: 440
indent: true
---
```

**Pull request merged**